bentoml/_internal/frameworks/catboost.py

```diff
@@ -0,0 +1,281 @@
+
+from ..utils.pkg import get_pkg_version
+from ..models.model import ModelContext
+
+if TYPE_CHECKING:
+    from bentoml.types import ModelSignature
+    from bentoml.types import ModelSignatureDict
+
+    from .. import external_typing as ext
+
+try:
+    import catboost as cb
+except ImportError:  # pragma: no cover
+    raise MissingDependencyException(
+        '"catboost" is required in order to use module `bentoml.catboost`, install '
+        "catboost with `pip install catboost`. For more information, refers to "
+        "https://catboost.ai/en/docs/concepts/installation"
+    )
+
+MODULE_NAME = "bentoml.catboost"
+MODEL_FILENAME = "saved_model.cbm"
+DEFAULT_MODEL_TRAINING_CLASS_NAME = "CatBoost"
+API_VERSION = "v1"
+
+logger = logging.getLogger(__name__)
+
+
+def get(tag_like: str | Tag) -> bentoml.Model:
+    """
+    Get the BentoML model with the given tag.
+
+    Args:
+        tag_like (``str`` ``|`` :obj:`~bentoml.Tag`):
+            The tag of the model to retrieve from the model store.
+
+    Returns:
+        :obj:`~bentoml.Model`: A BentoML :obj:`~bentoml.Model` with the matching tag.
+
+    Example:
+
+    .. code-block:: python
+
+        import bentoml
+        # target model must be from the BentoML model store
+        model = bentoml.catboost.get("my_catboost_model")
+    """
+    model = bentoml.models.get(tag_like)
+    if model.info.module not in (MODULE_NAME, __name__):
+        raise NotFound(
+            f"Model {model.tag} was saved with module {model.info.module}, not loading with {MODULE_NAME}."
+        )
+    return model
+
+
+def load_model(bento_model: str | Tag | bentoml.Model) -> cb.CatBoost:
+    """
+    Load the CatBoost model with the given tag from the local BentoML model store.
+
+    Args:
+        bento_model (``str`` ``|`` :obj:`~bentoml.Tag` ``|`` :obj:`~bentoml.Model`):
+            Either the tag of the model to get from the store, or a BentoML `~bentoml.Model`
+            instance to load the model from.
+
+    Returns:
+        :obj:`~catboost.CatBoost`: The CatBoost model loaded from the model store or BentoML :obj:`~bentoml.Model`.
+
+    Example:
+
+    .. code-block:: python
+
+        import bentoml
+        # target model must be from the BentoML model store
+        booster = bentoml.catboost.load_model("my_catboost_model")
+    """  # noqa: LN001
+    if not isinstance(bento_model, bentoml.Model):
+        bento_model = get(bento_model)
+
+    if bento_model.info.module not in (MODULE_NAME, __name__):
+        raise NotFound(
+            f"Model {bento_model.tag} was saved with module {bento_model.info.module}, not loading with {MODULE_NAME}."
+        )
+
+    model_file = bento_model.path_of(MODEL_FILENAME)
+    cb_class_name: str = bento_model.info.options.training_class_name  # type: ignore
+    cb_class: t.Type[cb.CatBoost] = getattr(cb, cb_class_name, cb.CatBoost)
+    if not issubclass(cb_class, cb.CatBoost) and cb_class is not cb.CatBoost:
```

Comment:
I think `issubclass` is reflexive.

```suggestion
    if not issubclass(cb_class, cb.CatBoost):
```